test/parallel/test-async-wrap-post-did-throw.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+require('../common');
+const assert = require('assert');
+const async_wrap = process.binding('async_wrap');
+var asyncThrows = 0;
+var ueCount = 0;
+
+process.on('uncaughtException', (e) => {
+  assert.equal(e.message, 'oh noes!', 'error messages do not match');
+});
+
+process.on('exit', () => {
+  process.removeAllListeners('uncaughtException');
+  assert.equal(ueCount, asyncThrows);
```

Comment:
Could you check that `ueCount` and `asyncThrows` equals 1. Just in case something about AsyncWrap causes `throw` not to propagate to the `post` hook and `uncaughtException` event.